### PR TITLE
Add support for repeating performances

### LINKED ISSUE #447: Allow for continuous, repeating Performances, without 'paused frames'
I'm working on an update to the CircularProgressIndicator to bring it in line with the Material Design spec (you can see the reasonably correct, although still not perfect Polymer implementation here: https://elements.polymer-project.org/elements/paper-spinner?view=demo:demo/index.html&active=paper-spinner)

This activity indicator will spin indefinitely until it's hidden or stopped.

While I haven't verified it, it appears in my current implementation there may be a 'paused frame' each time my performance is restarted.  By 'paused frame', I mean something different from a dropped frame, but rather a frame where the animation just stops progressing for a frame – when the performance is started again.

So what I'd like, is the ability to specify a number of iterations (which could be infinite) for a Performance, so that I can expect the animation to be perfectly smooth between the end state of one iteration and the start state of the next.

@Hixie and I discussed this a bit and may be able to provide more context

Below is some sample code of what I'm currently doing (I can point you at the full code if that is helpful).  Ideally I'd replace the 'Performance' with some new class... 'ContinuousPerformance'... or maybe I'd just use a Performance with a new argument 'iterations'?  And I'd be able to remove my status listener that restarts the Performance.

Please let me know if you need any more detail.

    _performance = new Performance(duration: new Duration(milliseconds: _kTotalDuration));

    _strokeHeadProgress = new AnimatedValue<double>(0.0, end: 1.0,
        curve: new Interval(0.0, 0.5, curve: Curves.fastOutSlowIn));

    _strokeTailProgress = new AnimatedValue<double>(0.0, end: 1.0,
            curve: new Interval(0.5, 1.0, curve: Curves.fastOutSlowIn));

    _circleRotation = new AnimatedValue<double>(0.0, end: 0.85, curve: Curves.linear);

    _stepIndex = 0;

    _performance.addStatusListener((PerformanceStatus status) {
      if (status == PerformanceStatus.completed) {
        _stepIndex++;
        if (_stepIndex >= 5) _stepIndex = 0;
        _restartAnimation();
      }
    });
    _performance.play();

### PR DESCRIPTION
**Pull request by abarth**
Fixes #447

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

**Comment by Hixie** [comment]
LGTM.

This code is so over-engineered. We really have to clean this up more. Just trying to review ten line patch I learnt about at least three new classes in our animation system that I didn't know about before.

**Comment by abarth** [comment]
Yeah...

File: packages/flutter/lib/src/animation/performance.dart [hunk 1]
```diff
@@ -249,6 +272,12 @@ class Performance extends PerformanceView {
     return _timeline.animateWith(force.release(progress, velocity));
   }
 
+  Future repeat({ double min: 0.0, double max: 1.0, Duration period }) {
+    if (period == null)
+      period = duration;
+    return _timeline.animateWith(new _RepeatingSimulation(min, max, period));
```
Comment (a14n) [inline packages/flutter/lib/src/animation/performance.dart]: You can use the _if null operator_ : `period ?? duration`

Comment (abarth) [inline packages/flutter/lib/src/animation/performance.dart]: Good point.

**Pull request merged**